generators/rabbitmq/files.mjs

```diff
@@ -10,39 +11,118 @@ import { moveToJavaPackageSrcDir } from '../server/support/index.mjs';
  */
 
 export const rabbitFiles = {
-    config: [
-      {
-        path: `${SERVER_MAIN_SRC_DIR}package/`,
-        renameTo: moveToJavaPackageSrcDir,
-        templates: ['config/RabbitMQConfig.java'],
-      },
-    ],
-    domain: [
-      {
-        path: `${SERVER_MAIN_SRC_DIR}package/`,
-        renameTo: moveToJavaPackageSrcDir,
-        templates: ['domain/RabbitMessageModel.java'],
-      },
-    ],
-    listener: [
-      {
-        path: `${SERVER_MAIN_SRC_DIR}package/`,
-        renameTo: moveToJavaPackageSrcDir,
-        templates: ['listener/RabbitMQConsumer.java'],
-      },
-    ],
-    publisher: [
+  config: [
+    {
+      path: `${SERVER_MAIN_SRC_DIR}package/`,
+      renameTo: moveToJavaPackageSrcDir,
+      templates: ['config/RabbitMQConfig.java'],
+    },
+  ],
+  domain: [
+    {
+      path: `${SERVER_MAIN_SRC_DIR}package/`,
+      renameTo: moveToJavaPackageSrcDir,
+      templates: ['domain/RabbitMessageModel.java'],
+    },
+  ],
+  listener: [
+    {
+      path: `${SERVER_MAIN_SRC_DIR}package/`,
+      renameTo: moveToJavaPackageSrcDir,
+      templates: ['listener/RabbitMQConsumer.java'],
+    },
+  ],
+  publisher: [
+    {
+      path: `${SERVER_MAIN_SRC_DIR}package/`,
+      renameTo: moveToJavaPackageSrcDir,
+      templates: ['publisher/RabbitMQPublisher.java'],
+    },
+  ],
+};
+
+export default async function writeRabbitMQFilesTask({ application }) {
+  if (communications.length === 0) {
+    application.queueName = application.baseName;
+    application.uniqueName = null;
+    this.writeFiles({
+      sections: rabbitFiles,
+      context: application,
+    });
+  } else {
+    for (let i = 0; i < communications.length; i++) {
+      if (this.jhipsterConfig.baseName === communications[i].client) {
+        const capitalizeServerName = communications[i].server.charAt(0).toUpperCase() + communications[i].server.slice(1);
+        const capitalizeClientName = communications[i].client.charAt(0).toUpperCase() + communications[i].client.slice(1);
+        const queueName = capitalizeServerName.concat('To').concat(capitalizeClientName);
+        this.fs.copyTpl(
+          this.templatePath('src/main/java/package/listener/RabbitMQConsumer.java.ejs'),
+          this.destinationPath(
+            `${SERVER_MAIN_SRC_DIR}`
+              .concat(this.jhipsterConfig.packageFolder)
+              .concat('/listener/RabbitMQConsumer'.concat(queueName).concat('.java'))
+          ),
+          {
+            packageName: this.jhipsterConfig.packageName,
+            queueName: queueName,
+            uniqueName: queueName,
+            serverName: communications[i].server,
+          }
+        );
+      }
+    }
+
+    for (let i = 0; i < communications.length; i++) {
+      if (this.jhipsterConfig.baseName === communications[i].server || communications.length === 0) {
```

Comment:
|| "communications.length === 0"
this condition is not required as the publisher files are already added  when the comminciations.length is equals to 0.